src/charm.py

```diff
@@ -69,6 +73,18 @@ def main(self, event):
 
         self.model.unit.status = ActiveStatus()
 
+    def _on_update_status(self, event):
+        try:
+            check_workload_health(
+                self._container, self._container_name, self._health_check_name, self.logger
+            )
+        except ErrorWithStatus as err:
+            self.model.unit.status = err.status
+            self.logger.error(f"Failed to handle {event} with error: {err}")
```

Comment:
Should we make this log a bit more explicit on where in the code flow it happened? What about:
```python
self.logger.error(f"Failed to handle {event} when checking workload helth: {err}")
```

Reply:
Not sure if this is really needed. This is supposedly a generic error handler (not workload health check specific). Right now though, this is only being used for checking workload health but this is still visible in the [error raised](https://github.com/canonical/charmed-kubeflow-chisme/pull/85/files#diff-50e457fc524ec1d8cdc688dbe62d4c42e2529929f3ba8bc690fde0a9f620ef2eR32) and the previous log too.
```
Workload failed health check
```